web/src/tippyjs.js

```diff
@@ -639,9 +639,16 @@ export function initialize() {
 
     delegate("body", {
         target: [".disabled-compose-send-button-container"],
-        content: $t({
-            defaultMessage: "You do not have permission to post in this stream.",
-        }),
+        content() {
+            if (compose_state.get_message_type() === "private") {
+                return $t({
+                    defaultMessage: "Direct messages are disabled in this organization.",
+                });
+            }
+            return $t({
+                defaultMessage: "You do not have permission to post in this stream.",
+            });
```

Comment:
It would be nice to avoid the code duplication that we have for these strings. One idea would be to extract a `compose_recipient.get_posting_policy_error_message` function that returns `""` or the error string depending on whether you're allowed to send messages, and use that both here and in  `check_posting_policy_for_compose_box`.

Reply:
Okay what I commented earlier was pending for 10 hours and now got deleted. 

I was saying that one thing to notice is I have created a new function in `stream_data` which is `get_sub_by_id`, to avoid 3 calls to get `can_post_messages_in_stream` as "get stream name", "get stream" and "get can_post_messages_in_stream", Now it is using 2 calls. 

Should I do this in a separate commit?